### PR TITLE
commit: Add --mode-ro-executables

### PR DESCRIPTION
**Pull request by cgwalters**
I think we should encourage removing the writable bits from
executables.  This has happened to me:
https://thomask.sdf.org/blog/2019/11/09/take-care-editing-bash-scripts.html

And not having the writable bit may help prevent hardlink
corruption with OSTree in some cases.

We can't do this by default, but add a convenient CLI flag
for it.

### CODE REVIEW

File: src/ostree/ot-builtin-commit.c [hunk 1]
```diff
@@ -111,6 +112,7 @@ static GOptionEntry options[] = {
   { "owner-uid", 0, 0, G_OPTION_ARG_INT, &opt_owner_uid, "Set file ownership user id", "UID" },
   { "owner-gid", 0, 0, G_OPTION_ARG_INT, &opt_owner_gid, "Set file ownership group id", "GID" },
   { "canonical-permissions", 0, 0, G_OPTION_ARG_NONE, &opt_canonical_permissions, "Canonicalize permissions in the same way bare-user does for hardlinked files", NULL },
+  { "mode-w-xor-x", 0, 0, G_OPTION_ARG_NONE, &opt_mode_w_xor_x, "Ensure executable files are not writable", NULL },
```
Comment (jlebon) [inline src/ostree/ot-builtin-commit.c:115]: Hmm, that's a bit cryptic (also I think it's technically a nand, not a xor :) ). Maybe something like `--ro-executables`?

File: tests/basic-test.sh [hunk 2]
```diff
@@ -541,6 +541,24 @@ fi
...
+cd ${test_tmpdir}
+rm test2-checkout -rf
+$OSTREE checkout test2 test2-checkout
+cd test2-checkout
+touch user-wx
+chmod 0755 user-wx
+touch group-wx
+chmod 0575 group-wx
+touch both-wx
+chmod 0775 both-wx
```
Comment (jlebon) [inline tests/basic-test.sh:553]: Maybe a sanity-check that a writable but not executable file doesn't get its write perms stripped?

File: src/ostree/ot-builtin-commit.c (hunk 1 above)
Comment (cgwalters) [reply src/ostree/ot-builtin-commit.c]: "xor" is about the "rule" and not the operation - https://en.wikipedia.org/wiki/W%5EX

Comment (jlebon) [reply src/ostree/ot-builtin-commit.c]: Ahh gotcha, didn't realize that.

**Pull request merged**